File: api/Models/Problem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Models
{
    public class Problem
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Difficulty { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public ICollection<TestCase> TestCases { get; set; } = new List<TestCase>();

    }
}

File: frontend/src/types/problem.ts
export interface TestCase {
  input: string;
  expectedOutput: string;
  isHidden: boolean;
}

export interface Problem {
  id: number;
  title: string;
  description: string;
  difficulty: "Fácil" | "Médio" | "Difícil";
  category: string;
  testCases: TestCase[];
}

export interface TestResult {
  testCase: TestCase;
  userOutput: string;
  passed: boolean;
  error?: string;
}

export interface SubmissionResult {
  passed: number;
  total: number;
  results: TestResult[];
  allPassed: boolean;
}
